src/eppo_value.ts

```diff
@@ -0,0 +1,39 @@
+export enum ValueType {
+  NullType,
+  BoolType,
+  NumericType,
+  StringType,
+}
+
+export type IValue = boolean | number | string | undefined;
+
+export class EppoValue {
+  public valueType: ValueType;
+  public boolValue: boolean | undefined;
+  public numericValue: number | undefined;
+  public stringValue: string | undefined;
+
+  constructor(
+    valueType: ValueType,
+    boolValue: boolean | undefined,
+    numericValue: number | undefined,
+    stringValue: string | undefined,
```

Comment:
If you want to enforce exactly one of the values is provided, you could do that with union types.

Although really I think we want this constructor to be `private` so people have to use the static factory methods.

Reply:
Good suggestion thank you.